lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -69,46 +104,211 @@ def status_code
+      # If fails, the error is added to the errors array
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @return [Boolean] True if the destination is valid, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate_destination(soft = true)
+        return true if destination.nil? || destination.empty? || settings.single_logout_service_url.nil? || settings.single_logout_service_url.empty?
 
-        unless self.options[:matches_request_id] == in_response_to
-          return soft ? false : validation_error("Response does not match the request ID, expected: <#{self.options[:matches_request_id]}>, but was: <#{in_response_to}>")
+        unless destination == current_url
+          error_msg = "The Logout Response was received at #{destination} instead of #{current_url}"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
         end
 
         true
       end
 
+      # Validates the Issuer of the Logout Response
+      # If fails, the error is added to the errors array
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @return [Boolean] True if the Issuer matchs the IdP entityId, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
       def valid_issuer?(soft = true)
-        return true if self.settings.idp_entity_id.nil? or self.issuer.nil?
+        return true if settings.idp_entity_id.nil? or issuer.nil?
 
-        unless URI.parse(self.issuer) == URI.parse(self.settings.idp_entity_id)
-          return soft ? false : validation_error("Doesn't match the issuer, expected: <#{self.settings.issuer}>, but was: <#{issuer}>")
+        unless URI.parse(issuer) == URI.parse(settings.idp_entity_id)
+          error_msg = "Doesn't match the issuer, expected: <#{settings.idp_entity_id}>, but was: <#{issuer}>"
+          @errors << error_msg
+          return soft ? false : validation_error(error_msg)
         end
+
         true
       end
+
+      # Validates the Signature if exists and GET parameters are provided
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @param get_params [Hash] In order to validate the In value we need to provide the GET parameters
+      # @return [Boolean] True if not contains a Signature or if the Signature is valid, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #      
+      def validate_signature(soft = true, get_params = nil)
+        return true if get_params.nil? || get_params['Signature'].nil? || settings.nil? || settings.get_idp_cert.nil?
```

Comment:
the `or` condition here should be moved into a boolean method with a descriptive name.